Iio regmap fw

tachicialex opened the pull request:
Changes regmap fw:
- moved the allocation of regmap_cfg inside iio_regmap_read_config()
- replaced parsing with sscanf. Now parsing is done with a strtok like function and kstrtoul
- moved all parsing in one function read_register_op
- added an array ,iio_regmap_op_required_fields, which specifies what a command should contain
- commands are now case insensitive and may have extra spaces

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -51,11 +119,265 @@ struct regmap_config *iio_regmap_read_config(struct device *dev)
+		return NULL;
+	}
+
+	reg_ops = devm_kzalloc(dev, new_lines * sizeof(*reg_ops), GFP_KERNEL);
```
you could just return directly here:  `return devm_kzalloc(dev, new_lines * sizeof(*reg_ops), GFP_KERNEL);`

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -51,11 +119,265 @@ struct regmap_config *iio_regmap_read_config(struct device *dev)
+	release_firmware(firmware);
+
+	reg_ops = alloc_register_ops(dev, input_fw, firmware->size);
+	if (!reg_ops) {
```
maybe use `IS_ERR(reg_ops)` and do `return ERR_CAST(reg_ops)`

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -51,11 +119,265 @@ struct regmap_config *iio_regmap_read_config(struct device *dev)
+	const char *c;
+
+	if (!fw_reg_ops)
+		return NULL;
```
maybe do `return ERR_PTR(-EINVAL)` here

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -51,11 +119,265 @@ struct regmap_config *iio_regmap_read_config(struct device *dev)
+
+	if (!new_lines) {
+		dev_err(dev, "Firmware file has an invalid format!\n");
+		return NULL;
```
maybe do `return ERR_PTR(-EINVAL)` here

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -404,9 +540,9 @@ int iio_regmap_probe(struct device *dev, struct regmap *regmap,
-	return ret;
+	return iio_run_regmap_ops(dev, regmap, register_ops, nr_ops);
 }
+
```
this line looks extra

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -404,9 +540,9 @@ int iio_regmap_probe(struct device *dev, struct regmap *regmap,
 		dev_err(dev, "parse_firmware failed!\n");
 		return PTR_ERR(register_ops);
 	}
-
```
no need to remove this line here;

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -312,6 +313,141 @@ static int parse_register_ops(struct device *dev, char *input_fw, int size_fw,
+	if (!reg_ops || !regmap)
+		return -EINVAL;
+
+	for (reg_op = reg_ops; reg_op < reg_ops + nr_ops; reg_op++) {
```
why not do 
```
for (i = 0; i < nr_ops; i++)  {
    reg_op = &reg_op[i] ;
.....
```

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -51,11 +119,265 @@ struct regmap_config *iio_regmap_read_config(struct device *dev)
+			return -EINVAL;
+		}
+		*input_fw_next_line = '\0';
+		ret = read_register_op(dev, input_fw, &reg_ops[nr_line],
```
`read_register_op` -> `parse_register_op`?

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -51,11 +119,265 @@ struct regmap_config *iio_regmap_read_config(struct device *dev)
+			return ret;
+	}
+	if (req_fields->dbg) {
+		ret = read_register_op_field(dev, &reg_op->dbg,
```
`read_register_op_field` -> `parse_register_op_field`?

commodo commented:
rest looks good

tachicialex commented:
Changes v2:
- alloc_register_ops now returns error codes with ERR_PTR
- renamed read_register_op_field -> parse_register_op_field
- renamed read_register_op -> parse_register_op
- iio_run_regmap_ops traverses reg_ops array using an index instead of pointer

commodo commented on drivers/iio/regmap/iio-regmap.c:
```diff
@@ -51,11 +119,259 @@ struct regmap_config *iio_regmap_read_config(struct device *dev)
+		return -EINVAL;
+	}
+
+	memset(reg_op_cmd, 0, REG_OP_STRLEN);
```
do you need to use the temp buffer here?
looks like if you remove it and use `fw_reg_cmd` all should be fine
